4MI0800264/Week 3/Problem 3.cpp

```diff
@@ -0,0 +1,101 @@
+#include <iostream>
+#include <fstream>
+using namespace std;
+
+struct Pair
+{
+	int a, b;
+};
+
+struct Relation
+{
+	Pair pairArray[25];
+	size_t size = 0;
+};
+
+Pair createPair(int x, int y)
+{
+	return { x,y };
+}
+
+void initPair(Pair& toInit, int x, int y)
+{
+	toInit.a = x;
+	toInit.b = y;
+}
+
+Pair readPairFromFile(std::ifstream& inFile)
+{
+	Pair newPair;
+
+	inFile >> newPair.a >> newPair.b;
+	inFile.close();
```

Comment:
Closing files is best handled in the same scope in which they got opened. This one here - this is a bad practice because what happens if one tries to invoke `readPairFromFile` twice ?